src/main/java/com/vagasti/backend/vaga/VagaController.java

```diff
@@ -0,0 +1,28 @@
+package com.vagasti.backend.vaga;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping(produces = "application/json")
+public class VagaController {
+  private final VagaService vagaService;
+
+  public VagaController(VagaService vagaService) {
+    this.vagaService = vagaService;
+  }
+
+  @PostMapping("/public/novaVaga")
```

Comment:
End point não pode ser público

```suggestion
  @PostMapping("/novaVaga")
```